openmoc/materialize.py

```diff
@@ -2,192 +2,319 @@
 # @file materialize.py
 # @package openmoc.materialize
 # @brief The materialize module provides utility functions to read and write
-#        multi-group materials cross-section data from HDF5 binary file format.
+#        multi-group materials cross section data from HDF5 binary file format.
 # @author William Boyd (wboyd@mit.edu)
-# @date April 23, 2013
-
-# Determine which OpenMOC module is being used. This is important for
-# the materialize module since it must instantiate Material objects
-# with the same floating point precision as that compiled into the
-# openmoc module used in the main Python input script to OpenMOC.
+# @date October 9, 2015
 
 import sys
+import os
+
+import numpy as np
```

Comment:
We did previously support an option to build OpenMOC without embedding the NumPy C API into the code. I gutted most of this functionality a while back (though it looks like I didn't *quite* finish the job then, so I did with my most recent commit here) since it was only there to support compilation on BGQ. Since BGQ really isn't (shouldn't in my opinion) be a target platform for us with the Python API, I think it is fine to make numpy a hard requirement since it is so widely available these days. And if anyone is interested in deploying the code on funky hardware like BGQ, they can simply use the Python-less alternative build system from #159.